### PR TITLE
Add embaas embeddings api endpoints 

### PR DESCRIPTION
**Pull request by juliuslipp**
# Introduces embaas embeddings endpoints

In this PR, we add support for embaas embeddings endpoints to Text Embedding Models (with LLMs, and Doc loaders in different PRs coming). We currently offer the MTEB leaderboard top performers, will continue to add top embedding models and soon add support for customers to deploy thier own models.Additional Documentation + Infomation can be found [here](https://embaas.io).

While developing this integration, I closely followed the patterns established by other langchain integrations. Nonetheless, if there are any aspects that require adjustments or if there's a better way to present a new integration, let me know! :)

#### Who can review?
  - @hwchase17
  - @agola11

### CODE REVIEW

File: langchain/embeddings/embaas.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+class EmbaasEmbeddings(BaseModel, Embeddings):
+    """Wrapper around Embaas's embedding service.
+
+    To use, you should have the
+    environment variable ``EMBAAS_API_KEY`` set with your API key, or pass
+    it as a named parameter to the constructor.
+
+    Example:
+        .. code-block:: python
```
Comment (eyurtsev) [inline langchain/embeddings/embaas.py:23]: A new line is recommended between the code block directive and the first code line (here a doc-string)

File: langchain/embeddings/embaas.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        return {"model": self.model, "instruction": self.instruction}
+
+    def _generate_payload(self, texts: List[str]) -> Dict[str, Any]:
+        """Generates payload for the API request."""
+        payload = {"texts": texts, "model": self.model}
+        if self.instruction:
+            payload["instruction"] = self.instruction
+        return payload
+
+    def _handle_request(self, payload: Dict[str, Any]) -> List[List[float]]:
```
Comment (eyurtsev) [inline langchain/embeddings/embaas.py:72]: (nit) up to you, but TypedDicts are nice type hints for this use case

**Comment by eyurtsev** [comment]
@juliuslipp thanks for the contribution. This code basically looks perfect. 

If you're able to resolve ilnt conflict, we'll be able to merge the code in faster.

Two things that would be nice to have (if you have time/desire):
1. A unit test that attempts to import the code
2. Or alternatively a unit test that uses the cod with [responses library](https://github.com/getsentry/responses). It'll make sure that this code will not break if other folks try to enhance it in the future.

**Comment by juliuslipp** [comment]
Hey @eyurtsev,

thanks for the quick review. I fixed your comments & the linting. 

About the tests, wasn't too sure what you meant by 1., but I freestyled a responses test and hope it works like this :D

**Pull request merged**